translates: forms

jochenboesmans opened the pull request:
TODO @Jacco 

- [x] get checks running good again
- [x] correct line numbers
- [ ] compare with original
- [ ] smaken -> nederlands
- [ ] approve
- [ ] get 2nd reviewer

jochenboesmans committed:
Translate forms.md

jochenboesmans committed:
Translate forms.md

jochenboesmans committed:
Update forms.md

jochenboesmans committed:
Update forms.md

Keraito commented:
Is this ready for review? If so, indicate it (using the labels) @jochenboesmans

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -9,7 +9,8 @@ redirect_from:
 ---
 
-HTML form elements work a little bit differently from other DOM elements in React, because form elements naturally keep some internal state. For example, this form in plain HTML accepts a single name:
+HTML form elementen werken een beetje anders dan andere DOM elementen in React omdat form elementen uit zichzelf een interne state bijhouden.
```
```suggestion
HTML form elementen werken een beetje anders dan andere DOM elementen in React omdat form elementen zelf een interne staat bijhouden.
```

`uit zichzelf` would be `automatically` and not `self / inside`

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -9,7 +9,8 @@ redirect_from:
 
-HTML form elements work a little bit differently from other DOM elements in React, because form elements naturally keep some internal state. For example, this form in plain HTML accepts a single name:
+HTML form elementen werken een beetje anders dan andere DOM elementen in React omdat form elementen uit zichzelf een interne state bijhouden.
+Bijvoorbeeld, onderstaande form in gewone HTML neemt een enkele naam aan:
```
```suggestion
Bijvoorbeeld, onderstaande `form` in gewone HTML neemt een enkele naam aan:
```

Either codify `form` or use `onderstaand formulier`

SleeplessByte commented on content/docs/forms.md:
````diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
 ```
 
-This form has the default HTML form behavior of browsing to a new page when the user submits the form. If you want this behavior in React, it just works. But in most cases, it's convenient to have a JavaScript function that handles the submission of the form and has access to the data that the user entered into the form. The standard way to achieve this is with a technique called "controlled comp ...
+Deze form gedraagt zich zoals een standaard HTML form voor het browsen naar een nieuwe pagina wanneer een gebruiker de form doorstuurt.
````
```suggestion
Deze `form` heeft het standaard HTML `form` gedrag van het browsen naar een nieuwe pagina wanneer een gebruiker de `form` verzend.
```

I would translate `submit` as `verzenden` and not `doorsturen` (send through / forward).

It's the behaviour `van het browsen` not `voor het browsen`

jochenboesmans replied:
Agreed this sentence is more clear, although it should be "verzendt".

```suggestion
Deze `form` heeft het standaard HTML `form` gedrag van het browsen naar een nieuwe pagina wanneer een gebruiker de `form` verzendt.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
-This form has the default HTML form behavior of browsing to a new page when the user submits the form. If you want this behavior in React, it just works. But in most cases, it's convenient to have a JavaScript function that handles the submission of the form and has access to the data that the user entered into the form. The standard way to achieve this is with a technique called "controlled comp ...
+Deze form gedraagt zich zoals een standaard HTML form voor het browsen naar een nieuwe pagina wanneer een gebruiker de form doorstuurt.
+Als je dit gedrag wilt in React, werkt het gewoon zo.
+Maar in de meeste gevallen, is het gemakkelijk om een JavaScript functie te hebben die het doorsturen van een form afhandelt en die toegang heeft tot de data die de gebruiker ingevuld heeft in de form.
```
```suggestion
Maar in de meeste gevallen is het handig om een JavaScript functie te hebben die het verzenden van een `form` afhandelt en die toegang heeft tot de data die de gebruiker in de `form` ingevuld heeft.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
+Deze form gedraagt zich zoals een standaard HTML form voor het browsen naar een nieuwe pagina wanneer een gebruiker de form doorstuurt.
+Als je dit gedrag wilt in React, werkt het gewoon zo.
+Maar in de meeste gevallen, is het gemakkelijk om een JavaScript functie te hebben die het doorsturen van een form afhandelt en die toegang heeft tot de data die de gebruiker ingevuld heeft in de form.
+De standaard manier om dit te implementeren is met een techniek die "gecontroleerde componenten" (in het Engels: "controlled components") wordt genoemd.
```
```suggestion
De standaard manier om dit te implementeren is met een techniek die "aangestuurde componenten" (in het Engels: "controlled components") wordt genoemd.
```

Other possible translations: `bestuurde componenten` `aangevoerde componenten` `beheerste componenten`

It's control in the sense of commanding / authority, not checking. `gecontroleerde` is therefore not the right translation.

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
+De standaard manier om dit te implementeren is met een techniek die "gecontroleerde componenten" (in het Engels: "controlled components") wordt genoemd.
 
-## Controlled Components {#controlled-components}
+## Gecontroleerde Componenten {#controlled-components}
```
See above

```suggestion
## Aangestuurde Componenten {#controlled-components}
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
+## Gecontroleerde Componenten {#controlled-components}
 
-In HTML, form elements such as `<input>`, `<textarea>`, and `<select>` typically maintain their own state and update it based on user input. In React, mutable state is typically kept in the state property of components, and only updated with [`setState()`](/docs/react-component.html#setstate).
+In HTML behouden form elementen zoals `<input>`, `<textarea>`, en `<select>` typisch hun eigen state en werken ze deze bij op basis van gebruikersinput.
```
```suggestion
In HTML behouden `form` elementen zoals `<input>`, `<textarea>`, en `<select>` typisch hun eigen staat en werken deze bij op basis van gebruikersinvoer.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
 
-In HTML, form elements such as `<input>`, `<textarea>`, and `<select>` typically maintain their own state and update it based on user input. In React, mutable state is typically kept in the state property of components, and only updated with [`setState()`](/docs/react-component.html#setstate).
+In HTML behouden form elementen zoals `<input>`, `<textarea>`, en `<select>` typisch hun eigen state en werken ze deze bij op basis van gebruikersinput.
+In React wordt veranderlijke state typisch bijgehouden in de state eigenschap van componenten en enkel bijgewerkt via de functie [`setState()`](/docs/react-component.html#setstate).
```
```suggestion
In React wordt veranderlijke staat typisch bijgehouden in de `state` eigenschap van componenten en enkel bijgewerkt via de functie [`setState()`](/docs/react-component.html#setstate).
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
+In React wordt veranderlijke state typisch bijgehouden in de state eigenschap van componenten en enkel bijgewerkt via de functie [`setState()`](/docs/react-component.html#setstate).
 
-We can combine the two by making the React state be the "single source of truth". Then the React component that renders a form also controls what happens in that form on subsequent user input. An input form element whose value is controlled by React in this way is called a "controlled component".
+We kunnen beiden combineren door de React state de "enkele bron van waarheid" te maken.
```
```suggestion
We kunnen beiden combineren door de React state de "enige bron van waarheid" te maken.
```

`Enkele` can also mean `a pair`, `enige` is always `single`

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
 
-We can combine the two by making the React state be the "single source of truth". Then the React component that renders a form also controls what happens in that form on subsequent user input. An input form element whose value is controlled by React in this way is called a "controlled component".
+We kunnen beiden combineren door de React state de "enkele bron van waarheid" te maken.
+Het React component dat een form rendert, beslist ook wat er gebeurt in die form bij volgende gebruikersinput.
```
```suggestion
In dat geval beslist het React component dat een form rendert ook wat er gebeurt in die form bij volgende gebruikersinvoer.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
-We can combine the two by making the React state be the "single source of truth". Then the React component that renders a form also controls what happens in that form on subsequent user input. An input form element whose value is controlled by React in this way is called a "controlled component".
+We kunnen beiden combineren door de React state de "enkele bron van waarheid" te maken.
+Het React component dat een form rendert, beslist ook wat er gebeurt in die form bij volgende gebruikersinput.
+Een input form element waarvan de value op deze manier gecontroleerd is door React wordt een "gecontroleerde component" genoemd.
```
```suggestion
Een `input` `form` element waarvan de `value` op deze manier aangestuurd is door React wordt een "aangestuurde component" genoemd.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -21,15 +22,21 @@ HTML form elements work a little bit differently from other DOM elements in Reac
+Een input form element waarvan de value op deze manier gecontroleerd is door React wordt een "gecontroleerde component" genoemd.
 
-For example, if we want to make the previous example log the name when it is submitted, we can write the form as a controlled component:
+Bijvoorbeeld, als we in het vorige voorbeeld de naam willen loggen in de console wanneer ze doorgestuurd wordt, kunnen we de form schrijven als een gecontroleerd component:
```
```suggestion
Bijvoorbeeld, als we in het vorige voorbeeld de naam willen loggen naar de console wanneer de naam verzonden wordt, kunnen we de `form` schrijven als een aangestuurd component:
```

Also wrong in the original text, `it` / `ze` refers to `console` and not `name` or `form input`

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -64,29 +71,33 @@ class NameForm extends React.Component {
+[**Probeer Het Op CodePen**](https://codepen.io/gaearon/pen/VmmPgp?editors=0010)
 
-Since the `value` attribute is set on our form element, the displayed value will always be `this.state.value`, making the React state the source of truth. Since `handleChange` runs on every keystroke to update the React state, the displayed value will update as the user types.
+Aangezien het `value` attribuut gezet is op ons form element, zal de weergegeven waarde altijd `this.state.value` zijn, waarbij de React state de enkele bron van waarheid is.
```
```suggestion
Aangezien het `value` attribuut meegegeven is aan ons `form` element, zal de weergegeven waarde altijd `this.state.value` zijn, waarbij de React staat de enige bron van waarheid is.
```

`gezet is op` is not really proper Dutch. You'll have to translate either in context of the component (`attributen worden meegegeven aan`) or in context of the JSX (`het attribute is toegevoegd aan`)

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -64,29 +71,33 @@ class NameForm extends React.Component {
 
-Since the `value` attribute is set on our form element, the displayed value will always be `this.state.value`, making the React state the source of truth. Since `handleChange` runs on every keystroke to update the React state, the displayed value will update as the user types.
+Aangezien het `value` attribuut gezet is op ons form element, zal de weergegeven waarde altijd `this.state.value` zijn, waarbij de React state de enkele bron van waarheid is.
+Aangezien `handleChange` loopt bij elke toetsaanslag om de React state bij te werken, zal de weergegeven waarde bijgewerkt worden wanneer de gebruiker typt.
```
```suggestion
Aangezien `handleChange` bij elke toetsaanslag wordt uitgevoerd om de React staat bij te werken, zal de weergegeven waarde bijgewerkt worden wanneer de gebruiker typt.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -64,29 +71,33 @@ class NameForm extends React.Component {
+Aangezien `handleChange` loopt bij elke toetsaanslag om de React state bij te werken, zal de weergegeven waarde bijgewerkt worden wanneer de gebruiker typt.
 
-With a controlled component, every state mutation will have an associated handler function. This makes it straightforward to modify or validate user input. For example, if we wanted to enforce that names are written with all uppercase letters, we could write `handleChange` as:
+Met een gecontroleerd component zal elke verandering van state een geassocieerde handler functie hebben.
```
```suggestion
Met een aangestuurd component zal elke verandering van de staat een bijbehorende _handler_ functie hebben.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -64,29 +71,33 @@ class NameForm extends React.Component {
 
-With a controlled component, every state mutation will have an associated handler function. This makes it straightforward to modify or validate user input. For example, if we wanted to enforce that names are written with all uppercase letters, we could write `handleChange` as:
+Met een gecontroleerd component zal elke verandering van state een geassocieerde handler functie hebben.
+Dit maakt het heel eenvoudig om gebruikersinput te veranderen of te valideren.
```
```suggestion
Dit maakt het heel eenvoudig om gebruikersinvoer te modificeren of te valideren.
```

I think `modificeren` works better here as `veranderen` would have been `change`

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -64,29 +71,33 @@ class NameForm extends React.Component {
-With a controlled component, every state mutation will have an associated handler function. This makes it straightforward to modify or validate user input. For example, if we wanted to enforce that names are written with all uppercase letters, we could write `handleChange` as:
+Met een gecontroleerd component zal elke verandering van state een geassocieerde handler functie hebben.
+Dit maakt het heel eenvoudig om gebruikersinput te veranderen of te valideren.
+Bijvoorbeeld, als we willen afdwingen dat namen geschreven worden met alleen hoofdletters, kunnen we `handleChange` als volgt schrijven:
```
```suggestion
Bijvoorbeeld, als we zouden willen afdwingen dat namen geschreven worden met alleen hoofdletters, zouden we `handleChange` als volgt kunnen schrijven:
```

It says `if we wanted ... we could write` and not `if we want to ... we can write`

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -64,29 +71,33 @@ class NameForm extends React.Component {
+## De textarea Tag {#the-textarea-tag}
 
-In HTML, a `<textarea>` element defines its text by its children:
+In HTML definieert een `<textarea>` element zijn tekst via zijn children:
```
```suggestion
In HTML definieert een `<textarea>` element zijn tekst via zijn `children`:
```

Or use `kinderen`

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -64,29 +71,33 @@ class NameForm extends React.Component {
 
-In React, a `<textarea>` uses a `value` attribute instead. This way, a form using a `<textarea>` can be written very similarly to a form that uses a single-line input:
+In React, daarentegen, gebruikt een `<textarea>` een `value` attribuut.
+Op deze manier kan een form dat een `<textarea>` gebruikt gelijkaardig worden geschreven als een form dat een enkele lijn input gebruikt.
```
```suggestion
Op deze manier kan een `form` dat een `<textarea>` gebruikt op gelijke manier worden geschreven als een `form` dat een `input` voor 1 regel tekst gebruikt.
```

Or at least:
```suggestion
Op deze manier kan een `form` dat een `<textarea>` gebruikt gelijkwaardig worden geschreven als een `form` dat een enkele lijn `input` gebruikt.
```

SleeplessByte commented on content/docs/forms.md:
````diff
@@ -123,11 +134,12 @@ class EssayForm extends React.Component {
 ```
 
-Notice that `this.state.value` is initialized in the constructor, so that the text area starts off with some text in it.
+Merk op dat `this.state.value` geïnitialiseerd werd in de constructor, waardoor de text area begint met wat tekst erin.
````
```suggestion
Merk op dat `this.state.value` geïnitialiseerd is in de `constructor`, waardoor de `textarea` direct wat tekst bevat.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -123,11 +134,12 @@ class EssayForm extends React.Component {
+## De select Tag {#the-select-tag}
 
-In HTML, `<select>` creates a drop-down list. For example, this HTML creates a drop-down list of flavors:
+In HTML creëert `<select>` een drop-down lijst.
```
```suggestion
In HTML creëert `<select>` een keuzelijst.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -123,11 +134,12 @@ class EssayForm extends React.Component {
 
-In HTML, `<select>` creates a drop-down list. For example, this HTML creates a drop-down list of flavors:
+In HTML creëert `<select>` een drop-down lijst.
+Bijvoorbeeld, deze HTML creëert een drop-down lijst van smaken:
```
```suggestion
Bijvoorbeeld, deze HTML creëert een keuzelijst van smaken:
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -138,7 +150,10 @@ In HTML, `<select>` creates a drop-down list. For example, this HTML creates a d
 
-Note that the Coconut option is initially selected, because of the `selected` attribute. React, instead of using this `selected` attribute, uses a `value` attribute on the root `select` tag. This is more convenient in a controlled component because you only need to update it in one place. For example:
+Merk op dat de Coconut optie initieel geselecteerd is omwille van het `selected` attribuut.
+React, daarentegen, gebruikt, in plaats van dit `selected` attribuut, een `value` attribuut op de root `select` tag.
```
```suggestion
React, gebruikt in plaats van dit `selected` attribuut, een `value` attribuut op de omvattende `select` tag.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -138,7 +150,10 @@ In HTML, `<select>` creates a drop-down list. For example, this HTML creates a d
-Note that the Coconut option is initially selected, because of the `selected` attribute. React, instead of using this `selected` attribute, uses a `value` attribute on the root `select` tag. This is more convenient in a controlled component because you only need to update it in one place. For example:
+Merk op dat de Coconut optie initieel geselecteerd is omwille van het `selected` attribuut.
+React, daarentegen, gebruikt, in plaats van dit `selected` attribuut, een `value` attribuut op de root `select` tag.
+Dit is gemakkelijker bij een gecontroleerd component omdat je het maar op één plaats moet bijwerken.
```
```suggestion
Dit is gemakkelijker bij een aangestuurde component omdat je het maar op één plaats moet bijwerken.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -178,33 +193,34 @@ class FlavorForm extends React.Component {
+[**Probeer Het Op CodePen**](https://codepen.io/gaearon/pen/JbbEzX?editors=0010)
 
-Overall, this makes it so that `<input type="text">`, `<textarea>`, and `<select>` all work very similarly - they all accept a `value` attribute that you can use to implement a controlled component.
+In het algemeen zorgt dit ervoor dat `<input type="text">`, `<textarea>`, en `<select>` allemaal op een zeer vergelijkbare manier functioneren - ze accepteren allemaal een `value` attribuut dat je kan gebruiken om een gecontroleerd component te implementeren.
```
```suggestion
In het algemeen zorgt dit ervoor dat `<input type="text">`, `<textarea>`, en `<select>` allemaal op een zeer vergelijkbare manier functioneren - ze accepteren allemaal een `value` attribuut dat je kan gebruiken om de component aan te sturen.
```

But a more correct translation would be 

```suggestion
In het algemeen zorgt dit ervoor dat `<input type="text">`, `<textarea>`, en `<select>` allemaal op een zeer vergelijkbare manier functioneren - ze accepteren allemaal een `value` attribuut dat je kan gebruiken om een aangestuurde component te implementeren.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -178,33 +193,34 @@ class FlavorForm extends React.Component {
+## De file input Tag {#the-file-input-tag}
 
-In HTML, an `<input type="file">` lets the user choose one or more files from their device storage to be uploaded to a server or manipulated by JavaScript via the [File API](https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications).
+In HTML, laat een `<input type="file">` de gebruiker één of meerdere bestanden kiezen uit de opslag van hun apparaat om geüpload te worden naar een server of gemanipuleerd te worden door JavaScript via de [File API](https://developer.mozilla.org/nl/docs/Web/API/File/Using_files_from_web_applications).
```
```suggestion
In HTML, laat een `<input type="file">` de gebruiker één of meerdere bestanden uit de opslag van hun apparaat kiezen om te uploaden naar een server of te manipuleren door JavaScript via de [File API](https://developer.mozilla.org/nl/docs/Web/API/File/Using_files_from_web_applications).
```

SleeplessByte commented on content/docs/forms.md:
````diff
@@ -178,33 +193,34 @@ class FlavorForm extends React.Component {
 ```
 
-Because its value is read-only, it is an **uncontrolled** component in React. It is discussed together with other uncontrolled components [later in the documentation](/docs/uncontrolled-components.html#the-file-input-tag).
+Omdat zijn waarde read-only is, is het een **ongecontroleerd** component in React.
````
```suggestion
Omdat zijn waarde alleen-lezen is, is het een **on-aangestuurd** component in React.
```

SleeplessByte replied:
But I would probably _want_ to say `kan je het component niet aansturen in React`

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -178,33 +193,34 @@ class FlavorForm extends React.Component {
 
-Because its value is read-only, it is an **uncontrolled** component in React. It is discussed together with other uncontrolled components [later in the documentation](/docs/uncontrolled-components.html#the-file-input-tag).
+Omdat zijn waarde read-only is, is het een **ongecontroleerd** component in React.
+Het wordt samen met ongecontroleerde componenten besproken [later in de documentatie](/docs/uncontrolled-components.html#the-file-input-tag).
```
```suggestion
Het wordt samen met andere on-aangestuurde componenten besproken [later in de documentatie](/docs/uncontrolled-components.html#the-file-input-tag).
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -178,33 +193,34 @@ class FlavorForm extends React.Component {
+## Meerdere Inputs Afhandelen {#handling-multiple-inputs}
 
-When you need to handle multiple controlled `input` elements, you can add a `name` attribute to each element and let the handler function choose what to do based on the value of `event.target.name`.
+Wanneer je meerdere gecontroleerde `input` elementen moet afhandelen, kan je een `name` attribuut toevoegen aan elk element en de handler functie laten kiezen wat te doen op basis van de waarde van `event.target.name`.
```
```suggestion
Wanneer je meerdere aangestuurde `input` elementen moet afhandelen, kan je een `name` attribuut toevoegen aan elk element en de `handler` functie laten kiezen wat te doen op basis van de waarde van `event.target.name`.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -254,31 +270,32 @@ class Reservation extends React.Component {
+[**Probeer Het Op CodePen**](https://codepen.io/gaearon/pen/wgedvV?editors=0010)
 
-Note how we used the ES6 [computed property name](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) syntax to update the state key corresponding to the given input name:
+Merk op hoe we de ES6 [computed property name](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) syntax gebruikt hebben om de state key overeenkomstig met de gegeven input name bij te werken:
```
```suggestion
Merk op hoe we de ES6 ["computed property name"](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) syntaxis gebruikt hebben om de sleutel in de staat overeenkomstig met de gegeven `input name` bij te werken:
```

SleeplessByte commented on content/docs/forms.md:
````diff
@@ -254,31 +270,32 @@ class Reservation extends React.Component {
 ```
 
-Also, since `setState()` automatically [merges a partial state into the current state](/docs/state-and-lifecycle.html#state-updates-are-merged), we only needed to call it with the changed parts.
+Ook, omdat `setState()` [automatisch gedeeltelijke state samenvoegt met de huidige state](/docs/state-and-lifecycle.html#state-updates-are-merged), moesten we deze functie enkel aanroepen met de veranderde delen.
````
```suggestion
Ook omdat `setState()` [automatisch onvolledige staat samenvoegt met de huidige state](/docs/state-and-lifecycle.html#state-updates-are-merged) hoefden we deze functie enkel aan te roepen met de veranderde delen.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -254,31 +270,32 @@ class Reservation extends React.Component {
+Ook, omdat `setState()` [automatisch gedeeltelijke state samenvoegt met de huidige state](/docs/state-and-lifecycle.html#state-updates-are-merged), moesten we deze functie enkel aanroepen met de veranderde delen.
 
-## Controlled Input Null Value {#controlled-input-null-value}
+## Gecontroleerde Input Null Value {#controlled-input-null-value}
```
```suggestion
## Aangestuurde Input "Null" Waarde {#controlled-input-null-value}
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -254,31 +270,32 @@ class Reservation extends React.Component {
+## Gecontroleerde Input Null Value {#controlled-input-null-value}
 
-Specifying the value prop on a [controlled component](/docs/forms.html#controlled-components) prevents the user from changing the input unless you desire so. If you've specified a `value` but the input is still editable, you may have accidentally set `value` to `undefined` or `null`.
+De value prop specificeren in een [gecontroleerd component](/docs/forms.html#controlled-components) weerhoudt de gebruiker van het veranderen van de input tenzij wanneer jij dit wilt.
```
```suggestion
De `value` prop specificeren in een [aangestuurd component](/docs/forms.html#controlled-components) weerhoudt de gebruiker van het veranderen van de `input` tenzij wanneer jij dit wilt.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -254,31 +270,32 @@ class Reservation extends React.Component {
 
-Specifying the value prop on a [controlled component](/docs/forms.html#controlled-components) prevents the user from changing the input unless you desire so. If you've specified a `value` but the input is still editable, you may have accidentally set `value` to `undefined` or `null`.
+De value prop specificeren in een [gecontroleerd component](/docs/forms.html#controlled-components) weerhoudt de gebruiker van het veranderen van de input tenzij wanneer jij dit wilt.
+Als je een `value` gespecificeerd hebt maar de input nog steeds bewerkbaar is, zou het kunnen dat je per ongeluk de `value` hebt gezet op `undefined` of `null`.
```
```suggestion
Als je een `value` gespecificeerd hebt maar de input nog steeds bewerkbaar is, zou het kunnen dat je per ongeluk de `value` de waarde `undefined` of `null` hebt gegeven.
```

SleeplessByte commented on content/docs/forms.md:
````diff
@@ -289,10 +306,13 @@ setTimeout(function() {
 ```
 
-## Alternatives to Controlled Components {#alternatives-to-controlled-components}
+## Alternatieven Voor Gecontroleerde Componenten {#alternatives-to-controlled-components}
````
```suggestion
## Alternatieven voor Aangestuurde Componenten {#alternatives-to-controlled-components}
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -289,10 +306,13 @@ setTimeout(function() {
+## Alternatieven Voor Gecontroleerde Componenten {#alternatives-to-controlled-components}
 
-It can sometimes be tedious to use controlled components, because you need to write an event handler for every way your data can change and pipe all of the input state through a React component. This can become particularly annoying when you are converting a preexisting codebase to React, or integrating a React application with a non-React library. In these situations, you might want to check out ...
+Het kan soms vervelend zijn om gecontroleerde componenten te gebruiken, omdat je een event handler moet schrijven voor elke manier waarop je data kan veranderen en alle input state doorheen het React component moet sturen.
```
```suggestion
Het kan soms vermoeiend zijn om aangestuurde componenten te gebruiken, omdat je een event handler moet schrijven voor elke manier waarop jouw data kan veranderen en alle input staat door het React component heen moet sturen.
```

Tedious was probably meant as monotone/fatiguing/dull. In this case `vermoeiend` is more appropriate.

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -289,10 +306,13 @@ setTimeout(function() {
 
-It can sometimes be tedious to use controlled components, because you need to write an event handler for every way your data can change and pipe all of the input state through a React component. This can become particularly annoying when you are converting a preexisting codebase to React, or integrating a React application with a non-React library. In these situations, you might want to check out ...
+Het kan soms vervelend zijn om gecontroleerde componenten te gebruiken, omdat je een event handler moet schrijven voor elke manier waarop je data kan veranderen en alle input state doorheen het React component moet sturen.
+Dit kan in het bijzonder vervelend worden wanneer je een bestaande codebase bent aan het omzetten naar React, of een React applicatie bent aan het integreren met een niet-React library.
```
```suggestion
Dit kan in het bijzonder vervelend worden wanneer je een bestaande _codebase_ naar React aan het omzetten bent, of een React applicatie aan het integreren bent met een niet-React library.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -289,10 +306,13 @@ setTimeout(function() {
-It can sometimes be tedious to use controlled components, because you need to write an event handler for every way your data can change and pipe all of the input state through a React component. This can become particularly annoying when you are converting a preexisting codebase to React, or integrating a React application with a non-React library. In these situations, you might want to check out ...
+Het kan soms vervelend zijn om gecontroleerde componenten te gebruiken, omdat je een event handler moet schrijven voor elke manier waarop je data kan veranderen en alle input state doorheen het React component moet sturen.
+Dit kan in het bijzonder vervelend worden wanneer je een bestaande codebase bent aan het omzetten naar React, of een React applicatie bent aan het integreren met een niet-React library.
+In deze situaties, wil je misschien [ongecontroleerde componenten](/docs/uncontrolled-components.html) bekijken als alternatieve techniek om input forms te implementeren.
```
```suggestion
In deze situaties wil je misschien [on-aangestuurde componenten](/docs/uncontrolled-components.html) bekijken als alternatieve techniek om _input forms_ te implementeren.
```

jochenboesmans replied:
```suggestion
In deze situaties wil je misschien [niet-aangestuurde componenten](/docs/uncontrolled-components.html) bekijken als alternatieve techniek om `input forms` te implementeren.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -289,10 +306,13 @@ setTimeout(function() {
+## Volwaardige Oplossingen {#fully-fledged-solutions}
 
-If you're looking for a complete solution including validation, keeping track of the visited fields, and handling form submission, [Formik](https://jaredpalmer.com/formik) is one of the popular choices. However, it is built on the same principles of controlled components and managing state — so don't neglect to learn them.
+Als je zoekt naar een volwaardige oplossing inclusief validatie, het bijhouden van bezoekte velden en het afhandelen van het doorsturen van forms, is [Formik](https://jaredpalmer.com/formik) één van de populaire keuzes.
```
```suggestion
Als je zoekt naar een volwaardige oplossing inclusief validatie, het bijhouden van bezochte velden en het afhandelen van het verzenden van `forms`, is [Formik](https://jaredpalmer.com/formik) één van de populaire keuzes.
```

SleeplessByte commented on content/docs/forms.md:
```diff
@@ -289,10 +306,13 @@ setTimeout(function() {
 
-If you're looking for a complete solution including validation, keeping track of the visited fields, and handling form submission, [Formik](https://jaredpalmer.com/formik) is one of the popular choices. However, it is built on the same principles of controlled components and managing state — so don't neglect to learn them.
+Als je zoekt naar een volwaardige oplossing inclusief validatie, het bijhouden van bezoekte velden en het afhandelen van het doorsturen van forms, is [Formik](https://jaredpalmer.com/formik) één van de populaire keuzes.
+Echter, het is gemaakt met dezelfde principes van gecontroleerde componenten en het beheren van state - het is dus zeker zinvol om deze te leren.
```
```suggestion
Het is echter gemaakt met dezelfde principes van aangestuurde componenten en het beheren van staat - het is dus zeker zinvol om daarover te leren.
```

jochenboesmans committed:
Codified use of "form" in Dutch translation.

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Codified use of "form" in Dutch translation.

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

jochenboesmans committed:
Update content/docs/forms.md

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

jochenboesmans committed:
Update content/docs/forms.md

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

jochenboesmans committed:
Update content/docs/forms.md

Co-Authored-By: Derk-Jan Karrenbeld <derk-jan+github@karrenbeld.info>

SleeplessByte commented:
@jochenboesmans I hope my suggestions made sense!

Commit:
Update lists-and-keys.md: eliminate huge space between properties (#2980)

jaysoo committed:
Add a link to Nx under the Flexible Toolchains section (#2979)

Jacco committed:
Merge branch 'master' into translate_forms

Commit:
few line feeds, will checks work again?

Commit:
few line feeds, will checks work again?

Commit:
lines moved together again

Commit:
lines moved together again

Commit:
v1.3.2

Commit:
v1.3.2

Viral-Inc committed:
2 Typos (#2988)

minor

macginitie committed:
corrected preposition in tutorial.md (#2986)

nickserv committed:
Fix incorrect section and out-of-date info for "Road to React" course (#2968)

sijusamson committed:
[Documentation] [Main concepts: State and Lifecycle] (#2981)

Co-authored-by: R Nabors <rachelnabors@users.noreply.github.com>

Jacco commented on content/docs/forms.md:
```diff
@@ -21,15 +21,15 @@ HTML form elements work a little bit differently from other DOM elements in Reac
+Deze `form` heeft het standaard HTML `form` gedrag van het browsen naar een nieuwe pagina wanneer een gebruiker de `form` verzendt. Als je dit gedrag wilt in React, werkt het gewoon zo. Maar in de meeste gevallen is het handig om een JavaScript functie te hebben die het verzenden van een `form` afhandelt en die toegang heeft tot de data die de gebruiker in de `form` ingevuld heeft. De standaard m ...
 
-## Controlled Components {#controlled-components}
+## Gecontroleerde Componenten {#controlled-components}
```
```suggestion
## Controlled Componenten {#controlled-components}
```

Jacco commented on content/docs/forms.md:
```diff
@@ -9,7 +9,7 @@ redirect_from:
 ---
 
-HTML form elements work a little bit differently from other DOM elements in React, because form elements naturally keep some internal state. For example, this form in plain HTML accepts a single name:
+HTML form elementen werken een beetje anders dan andere DOM elementen in React omdat form elementen zelf een interne staat bijhouden. Bijvoorbeeld, onderstaande `form` in gewone HTML neemt een enkele naam aan:
```
```suggestion
HTML-form-elementen werken in React een beetje anders dan andere DOM-elementen, omdat form-elementen van nature een interne staat bijhouden. Het onderstaande formulier in gewone HTML bijvoorbeeld, neemt een enkele naam aan:
```

Jacco commented on content/docs/forms.md:
```diff
@@ -248,31 +248,31 @@ class Reservation extends React.Component {
+## Gecontroleerde Input Null Value {#controlled-input-null-value}
 
-Specifying the value prop on a [controlled component](/docs/forms.html#controlled-components) prevents the user from changing the input unless you desire so. If you've specified a `value` but the input is still editable, you may have accidentally set `value` to `undefined` or `null`.
+De value prop specificeren in een [gecontroleerd component](/docs/forms.html#controlled-components) weerhoudt de gebruiker van het veranderen van de input tenzij wanneer jij dit wilt. Als je een `value` gespecificeerd hebt maar de input nog steeds bewerkbaar is, zou het kunnen dat je per ongeluk de `value` hebt gezet op `undefined` of `null`.
```
```suggestion
De value prop specificeren in een [controlled component](/docs/forms.html#controlled-components) weerhoudt de gebruiker van het veranderen van de input behalve wanneer jij dat wilt. Als je een `value` gespecificeerd hebt maar de input is nog steeds bewerkbaar, zou het kunnen dat je per ongeluk de `value` op `undefined` of `null` hebt gezet.
```

Jacco commented on content/docs/forms.md:
````diff
@@ -283,10 +283,10 @@ setTimeout(function() {
 ```
 
-## Alternatives to Controlled Components {#alternatives-to-controlled-components}
+## Alternatieven Voor Gecontroleerde Componenten {#alternatives-to-controlled-components}
````
```suggestion
## Alternatieven voor Controlled Components {#alternatives-to-controlled-components}
```

Jacco commented on content/docs/forms.md:
```diff
@@ -248,31 +248,31 @@ class Reservation extends React.Component {
+[**Probeer Het Op CodePen**](https://codepen.io/gaearon/pen/wgedvV?editors=0010)
 
-Note how we used the ES6 [computed property name](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) syntax to update the state key corresponding to the given input name:
+Merk op hoe we de ES6 [computed property name](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) syntax gebruikt hebben om de state key overeenkomstig met de gegeven input name bij te werken:
```
```suggestion
Merk op hoe we de ES6 ["computed property name"](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) syntax gebruikt hebben om de state key die hoort bij de naam input bij te werken:
```

Jacco commented on content/docs/forms.md:
```diff
@@ -248,31 +248,31 @@ class Reservation extends React.Component {
+Ook, omdat `setState()` [automatisch gedeeltelijke state samenvoegt met de huidige state](/docs/state-and-lifecycle.html#state-updates-are-merged), moesten we deze functie enkel aanroepen met de veranderde delen.
 
-## Controlled Input Null Value {#controlled-input-null-value}
+## Gecontroleerde Input Null Value {#controlled-input-null-value}
```
```suggestion
## Controlled Input Null Waarde {#controlled-input-null-value}
```

Jacco commented on content/docs/forms.md:
```diff
@@ -283,10 +283,10 @@ setTimeout(function() {
+## Alternatieven Voor Gecontroleerde Componenten {#alternatives-to-controlled-components}
 
-It can sometimes be tedious to use controlled components, because you need to write an event handler for every way your data can change and pipe all of the input state through a React component. This can become particularly annoying when you are converting a preexisting codebase to React, or integrating a React application with a non-React library. In these situations, you might want to check out ...
+Het kan soms vermoeiend zijn om aangestuurde componenten te gebruiken, omdat je een event handler moet schrijven voor elke manier waarop jouw data kan veranderen en alle input staat door het React component heen moet sturen. Dit kan in het bijzonder vervelend worden wanneer je een bestaande _codebase_ naar React aan het omzetten bent, of een React applicatie aan het integreren bent met een niet-R ...
```
```suggestion
Het kan soms vervelend zijn om aangestuurde componenten te gebruiken, omdat je een event handler moet schrijven voor elke manier waarop jouw data kan veranderen, en alle input-state door de React component heen moet sturen. Dit kan in het bijzonder irritant worden wanneer je een bestaande _codebase_ naar React aan het omzetten bent, of een React applicatie aan het integreren bent met een niet op React gebaseerde bibliotheek. In deze situaties wil je misschien [niet-aangestuurde componenten](/docs/uncontrolled-components.html) bekijken als alternatieve techniek om `input forms` te implementeren.
```

Jacco commented on content/docs/forms.md:
```diff
@@ -21,15 +21,15 @@ HTML form elements work a little bit differently from other DOM elements in Reac
+## Gecontroleerde Componenten {#controlled-components}
 
-In HTML, form elements such as `<input>`, `<textarea>`, and `<select>` typically maintain their own state and update it based on user input. In React, mutable state is typically kept in the state property of components, and only updated with [`setState()`](/docs/react-component.html#setstate).
+In HTML behouden form elementen zoals `<input>`, `<textarea>`, en `<select>` typisch hun eigen state en werken ze deze bij op basis van gebruikersinput. In React wordt veranderlijke state typisch bijgehouden in de state eigenschap van componenten en enkel bijgewerkt via de functie [`setState()`](/docs/react-component.html#setstate).
```
```suggestion
In HTML beheren form-elementen zoals `<input>`, `<textarea>`, en `<select>` normaal gesproken hun eigen state, en werken ze deze bij op basis van gebruikersinvoer. In React wordt veranderlijke state gewoonlijk bijgehouden in de state eigenschap van componenten en enkel bijgewerkt via de functie [`setState()`](/docs/react-component.html#setstate).
```

Jacco commented on content/docs/forms.md:
````diff
@@ -117,11 +117,11 @@ class EssayForm extends React.Component {
 ```
 
-Notice that `this.state.value` is initialized in the constructor, so that the text area starts off with some text in it.
+Merk op dat `this.state.value` geïnitialiseerd werd in de constructor, waardoor de text area begint met wat tekst erin.
````
```suggestion
Merk op dat `this.state.value` geïnitialiseerd is in de `constructor`, waardoor het tekstvak direct wat tekst bevat.
```

Jacco commented on content/docs/forms.md:
```diff
@@ -117,11 +117,11 @@ class EssayForm extends React.Component {
+## De select Tag {#the-select-tag}
 
-In HTML, `<select>` creates a drop-down list. For example, this HTML creates a drop-down list of flavors:
+In HTML creëert `<select>` een drop-down lijst. Bijvoorbeeld, deze HTML creëert een drop-down lijst van smaken:
```
```suggestion
In HTML creëert `<select>` een drop-down lijst. Deze HTML creëert bijvoorbeeld een drop-down lijst van smaken:
```

Jacco commented on content/docs/forms.md:
````diff
@@ -132,7 +132,7 @@ In HTML, `<select>` creates a drop-down list. For example, this HTML creates a d
 ```
 
-Note that the Coconut option is initially selected, because of the `selected` attribute. React, instead of using this `selected` attribute, uses a `value` attribute on the root `select` tag. This is more convenient in a controlled component because you only need to update it in one place. For example:
+Merk op dat de Coconut optie initieel geselecteerd is omwille van het `selected` attribuut. React, daarentegen, gebruikt, in plaats van dit `selected` attribuut, een `value` attribuut op de root `select` tag. Dit is gemakkelijker bij een gecontroleerd component omdat je het maar op één plaats moet bijwerken. Bijvoorbeeld:
````
```suggestion
Merk op dat de Coconut optie in eerste instantie geselecteerd is vanwege het attribuut `selected`. React gebruikt, in plaats van dit `selected`-attribuut, een `value`-attribuut op de root tag `select`. Dit is handiger bij een aangestuurd component omdat je het maar op één plaats hoeft bij te werken. Bijvoorbeeld:
```

Jacco commented on content/docs/forms.md:
````diff
@@ -172,33 +172,33 @@ class FlavorForm extends React.Component {
 ```
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/JbbEzX?editors=0010)
+[**Probeer Het Op CodePen**](https://codepen.io/gaearon/pen/JbbEzX?editors=0010)
````
```suggestion
[**Probeer het op CodePen**](https://codepen.io/gaearon/pen/JbbEzX?editors=0010)
```

Commit:
Oops, was trying to get checks working again

Commit:
Oops, was trying to get checks working again

Commit:
trying to fix build failure

Commit:
trying to fix build failure

Commit:
revert version (didn't help)

Commit:
revert version (didn't help)

Jacco commented:
@tesseralis something is wrong with the netlify checks.

Some how too netlify deploys are happening (one is owned by you @Keraito : frosty-bohr and now also mine infallible-tesla)

Anyway all three builds fail with an error somewhere in the logs:
```
8:10:54 AM: info bootstrap finished - 6.373 s
8:10:54 AM: ⠀
8:10:55 AM: 
8:10:55 AM: failed Building production JavaScript and CSS bundles - 0.655s
8:10:55 AM: 
8:10:55 AM: error Generating JavaScript bundles failed
8:10:55 AM: Cannot find module '@babel/compat-data/corejs3-shipped-proposals'
8:10:55 AM: Require stack:
8:10:55 AM: - /opt/build/repo/node_modules/@babel/preset-env/lib/polyfills/corejs3/usage-plugin.js
8:10:55 AM: - /opt/build/repo/node_modules/@babel/preset-env/lib/index.js
8:10:55 AM: - /opt/build/repo/node_modules/@babel/core/lib/config/files/plugins.js
8:10:55 AM: - /opt/build/repo/node_modules/@babel/core/lib/config/files/index.js
8:10:55 AM: - /opt/build/repo/node_modules/@babel/core/lib/index.js
8:10:55 AM: - /opt/build/repo/node_modules/babel-loader/lib/index.js
8:10:55 AM: - /opt/build/repo/node_modules/gatsby/dist/utils/babel-loader.js
8:10:55 AM: - /opt/build/repo/node_modules/loader-runner/lib/loadLoader.js
8:10:55 AM: - /opt/build/repo/node_modules/loader-runner/lib/LoaderRunner.js
8:10:55 AM: - /opt/build/repo/node_modules/webpack/lib/NormalModule.js
8:10:55 AM: - /opt/build/repo/node_modules/webpack/lib/NormalModuleFactory.js
8:10:55 AM: - /opt/build/repo/node_modules/webpack/lib/Compiler.js
8:10:55 AM: - /opt/build/repo/node_modules/webpack/lib/webpack.js
8:10:55 AM: - /opt/build/repo/node_modules/gatsby/dist/commands/build-html.js
8:10:55 AM: - /opt/build/repo/node_modules/gatsby/dist/commands/build.js
8:10:55 AM: - /opt/build/repo/node_modules/gatsby-cli/lib/create-cli.js
8:10:55 AM: - /opt/build/repo/node_modules/gatsby-cli/lib/index.js
8:10:55 AM: - /opt/build/repo/node_modules/gatsby/dist/bin/gatsby.js
8:10:55 AM: not finished run queries - 0.664s
8:10:55 AM: ​
8:10:55 AM: ┌─────────────────────────────┐
8:10:55 AM: │   "build.command" failed    │
8:10:55 AM: └─────────────────────────────┘
8:10:55 AM: ​
8:10:55 AM:   Error message
8:10:55 AM:   Command failed with exit code 1: gatsby build
8:10:55 AM: ​
8:10:55 AM:   Error location
8:10:55 AM:   In Build command from settings:
8:10:55 AM:   gatsby build
8:10:55 AM: ​
8:10:55 AM:   Resolved config
8:10:55 AM:   build:
8:10:55 AM:     command: gatsby build
8:10:55 AM:     publish: /opt/build/repo/public
8:10:55 AM: Caching artifacts
```

threepointone committed:
#blacklivesmatter (#3004)

gaearon committed:
Try to fix build

gaearon committed:
Add a BLM banner

gaearon committed:
CSS fixes for BLM styling

gaearon committed:
Fix flow

rickhanlonii committed:
Clarify language in components-and-props.md (#3020)

* Clarify language in components-and-props.md

This is a nit for a few reasons:
- The point that extracting a component that "is complex enough on its own" means that it's not necessarily reusable.
- Extracting the components listed in the same sentence (e.g. App) are not necessarily reusable.
- This ends the section "Extracting Components" with a reference to extraction instead of re-usability.
- This delimits extracting into a separate component from creating a re-usable component.

* Update components-and-props.md

* Update components-and-props.md

threepointone committed:
remove grayscale treatment (#3024)

As per https://twitter.com/sliminality/status/1269033413488459776, there's feedback that the contrast isn't great for reading. I myself have trouble with low contrast text, (and long sightedness), and I thought it was fine. I was clearly mistaken. Removing this for now to get back to a non harmful position, and we can decide later what to do instead.

reactjs-translation-bot committed:
merging all conflicts

Jacco committed:
Fixed conflict

Jacco committed:
Update content/docs/forms.md

Hopelijk doet netlify het weer?

Jacco committed:
Update content/docs/forms.md

Hopelijk doet netlify het weer?

Jacco committed:
Merge pull request #78 from reactjs/sync-c9b99007

Sync with reactjs.org @ c9b99007

Commit:
Merge branch 'translate_forms' of https://github.com/reactjs/nl.reactjs.org into translate_forms
